ci/run-wasi-nn-example.sh

```diff
@@ -18,24 +19,40 @@ else
     REMOVE_TMP_DIR=0
 fi
 
-# Build Wasmtime with wasi-nn enabled; we attempt this first to avoid extra work if the build fails.
+# One of the examples expects to be in a specifically-named directory.
+mkdir -p $TMP_DIR/mobilenet
+TMP_DIR=$TMP_DIR/mobilenet
+
+# Build Wasmtime with wasi-nn enabled; we attempt this first to avoid extra work
+# if the build fails.
 cargo build -p wasmtime-cli --features wasi-nn
 
 # Download all necessary test fixtures to the temporary directory.
 wget --no-clobber $FIXTURE/mobilenet.bin --output-document=$TMP_DIR/model.bin
 wget --no-clobber $FIXTURE/mobilenet.xml --output-document=$TMP_DIR/model.xml
 wget --no-clobber $FIXTURE/tensor-1x224x224x3-f32.bgr --output-document=$TMP_DIR/tensor.bgr
 
-# Now build an example that uses the wasi-nn API.
+# Now build an example that uses the wasi-nn API. Run the example in Wasmtime
+# (note that the example uses `fixture` as the expected location of the
+# model/tensor files).
 pushd $WASMTIME_DIR/crates/wasi-nn/examples/classification-example
 cargo build --release --target=wasm32-wasi
 cp target/wasm32-wasi/release/wasi-nn-example.wasm $TMP_DIR
 popd
+cargo run -- run --mapdir fixture::$TMP_DIR \
+    --wasi-modules=experimental-wasi-nn $TMP_DIR/wasi-nn-example.wasm
 
-# Run the example in Wasmtime (note that the example uses `fixture` as the expected location of the model/tensor files).
-cargo run -- run --mapdir fixture::$TMP_DIR --wasi-modules=experimental-wasi-nn $TMP_DIR/wasi-nn-example.wasm
+# Build and run another example, this time using Wasmtime's --graph flag to
+# preload the model.
+pushd $WASMTIME_DIR/crates/wasi-nn/examples/classification-example-named
+cargo build --release --target=wasm32-wasi
+cp target/wasm32-wasi/release/wasi-nn-example-named.wasm $TMP_DIR
+popd
+cargo run -- run --mapdir fixture::$TMP_DIR --graph openvino::$TMP_DIR \
+    --wasi-modules=experimental-wasi-nn $TMP_DIR/wasi-nn-example-named.wasm
```

Comment:
I'm not quite happy with using this script for testing. It was good initially to get things working but I would prefer to be using `cargo test` instead. The problem is that there is a dependency expectation: this can only run if the system has OpenVINO installed, which will not always be the case for developers running `cargo test`.

I can think of two workarounds:
- `#[ignore]` the OpenVINO-specific tests and add some comments telling developers to turn them on to fully test the system
- add some checks at the beginning of each test to early-exit if OpenVINO is not available; the issue with this is that users may see "false success" if something goes wrong in the environment, so perhaps a `FORCE_RUN_OPENVINO_TESTS` env variable is necessary?

Interested in some feedback on whether one of those approaches is better than this script!

Reply:
Personally I like your second suggestion and matches what I was going to suggest as well. We do run a higher risk of not actually running the tests in CI but I think running something by default as part of `cargo test --workspace` is a good mitigating factor for that.

Reply:
If the tests are for wasi-nn, we should just have a test backend that does something basic like an affine layer. 

If this is testing integration we should think about how we handle multiple backends. This same issue will come up with pytorch, onnxruntime, tf, etc.

Reply:
@alexcrichton, I have a whole other set of commits ready for switching to that kind of testing. Just waiting on a review of this one because the new testing also covers the `load_by_name` logic added here.

Reply:
Ok, check out https://github.com/bytecodealliance/wasmtime/pull/6895 if you're interested in where I went with this.